.github/workflows/nightly-pypi-build.yml

```diff
@@ -0,0 +1,89 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: "Nightly PyPI Build"
+
+on:
+  schedule:
+    - cron: "0 0 * * *"  # Runs at midnight UTC every day
+  workflow_dispatch:  # Allows manual triggering
+
+jobs:
+  set-version:
+    if: github.repository == 'apache/iceberg-python'  # Only run for apache repo
+    runs-on: ubuntu-latest
+    outputs:
+      VERSION: ${{ steps.set-version.outputs.VERSION }}
+      RC: ${{ steps.set-version.outputs.RC }}
```

Comment:
I find the term `RC` confusing here. How about simplifying this by not allow the user to set this, and also include the time. This is similar to Java: https://repository.apache.org/content/groups/snapshots/org/apache/iceberg/iceberg-core/1.8.0-SNAPSHOT/

This way we don't have to set this manually.